web/modules/weather_data/src/Service/Test/Base.php.test

```diff
@@ -79,6 +94,10 @@ abstract class Base extends TestCase
 
         $this->cacheMock = $this->createMock(CacheBackendInterface::class);
         $this->requestMock = $this->createStub(Request::class);
+        $this->databaseMock = $this->createStub(Connection::class);
+        $this->selfMock = $this->getMockBuilder(WeatherDataService::class)
+            ->disableOriginalConstructor()
+            ->getMock();
```

Comment:
You'll see below how this gets used, but in a nutshell, it's a way for us to isolate one method from all the others in the service. It should mean less mocking HTTP responses.